### ISSUE TITLE
TERM signal not working

### ISSUE THREAD

**Issue by MrMMorris** [hw-cookbooks/runit#67]
So I am trying to send Sidekiq a TERM signal using runit at the end of a deploy. At the beginning of the deploy, I send it a USR1 which works fine which is confirmed by the process showing "stopping". With the TERM signal, however, nothing happens. 

Recipe:
```ruby
action :before_migrate do
  runit_service 'sidekiq' do
    action :usr1 # WORKS
  end
end

action :after_restart do
  runit_service 'sidekiq' do
    action :term # DOESN'T WORK
  end
end
```

Chef shows:
```
runit_service[sidekiq] action term
    - send term to runit_service[sidekiq]
```

but the process still shows "stopping" as if no further signal was received after USR1. 

If I manually send `sudo sv TERM sidekiq` I get:

`kill: run: sidekiq: (pid 7263) 3176s, got TERM`

and the process restarts. The `runit_service` resource only accepts `action :term`, not `action :TERM` so **I'm not sure if the conversion of the parameter action to the actual signal is the issue**

I have also just tried referencing the init.d symlink:
```
execute '/etc/init.d/sidekiq TERM' do
  user 'root'
end
```

but it seems to result in a non-zero exit code:

```
Mixlib::ShellOut::ShellCommandFailed
  ------------------------------------
  execute[/etc/init.d/sidekiq TERM] (/var/chef/cache/cookbooks/application_ruby/providers/sidekiq.rb line 74) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
  ---- Begin output of /etc/init.d/sidekiq TERM ----
  STDOUT: kill: run: sidekiq: (pid 19404) 734s, got TERM
  STDERR:
  ---- End output of /etc/init.d/sidekiq TERM ----
  Ran /etc/init.d/sidekiq TERM returned 1
```

I found an answer stating that for Capistrano deploys they would stop the sidekiq runit service, manually stop sidekiq and then start the runit sidekiq service, but that seems super hacky :grimacing:

http://stackoverflow.com/questions/22060610/sidekiq-fails-to-stop-under-runit/22077734#22077734

**Any idea why the term signal doesn't work?**

**Comment by cwjohnston** [comment]
Today I learned something new about the term behavior for `sv`. In hopes of reproducing the behavior described here, I configured a service which runs [this stackoverflow script](http://stackoverflow.com/a/2183063) for logging the received signals. 

From testing these two variations of the `sv` command, I've observed that the lowercase `sv term servicename` invocation does in fact issue a TERM signal, but, in doing so, sv does not wait for the process to exit before returning with exit status 0. The uppercase `sv TERM servicename` variation also sends a TERM signal, but in this invocation sv seems to wait until the process has exited or timeout is reached and the process is killed.

@MrMMorris is it possible that what you observed in this issue can be attributed to sidekiq taking longer than expected to exit after receiving a TERM? Are you able to reproduce this issue under a current release?
